### ISSUE TITLE
[kube-prometheus-stack] Split CRDs in seperate chart (there as template)

### ISSUE THREAD

**Issue by rufreakde** [prometheus-community/helm-charts#3038]
### Is your feature request related to a problem ?

Declarative Automation CD tools suffer from the problem of not being able to annotate/Label CRDs (e.g. https://github.com/prometheus-community/helm-charts/issues/1500) 

Allowing templating of the CRDs `metadata.` section. This can easily solve this issue.

### Describe the solution you'd like.

The solution as described by HELM Documentation would be to split out the central CRD files into its own chart.
https://helm.sh/docs/chart_best_practices/custom_resource_definitions/#method-2-separate-charts

This workflow is perfect for Administrator tasks. And installing Prometheus is IMHO an Administrator task.

### Describe alternatives you've considered.

Allowing metadata configuration from Helm3 side to avoid this altogether but this is currently not possible.
https://github.com/helm/helm/pull/11022 

### Additional context.

It seems the chart maintainers are also aware of this issue with the CRDs and their sizes and update problems hence there is this section of `kubectl apply` commands to mitigate this problem Manualy. 
https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#from-44x-to-45x

e.g. 
```yaml
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.63.0/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml
...
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.63.0/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
kubectl apply --server-side -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/v0.63.0/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
```

**Note** please notice the `--server-side` which is important for this CRDs as they are rather big.

**Comment by zeritti** [comment]
The prometheus-operator-crds chart has been made available through PR #2697 (more info in issue #2612).

**Comment by rufreakde** [comment]
Thanks for sharing. Could maybe make this more prominent in the readme as an installation step. Instead of the kubectl apply workarounds! At least starting from the newest versions the recommended way could be 2 charts installation to avoid the problems.
